go/vt/vtgate/planbuilder/dml.go

```diff
@@ -39,7 +39,17 @@ func buildUpdatePlan(upd *sqlparser.Update, vschema VSchema) (*engine.Route, err
 	route := &engine.Route{
 		Query: generateQuery(upd),
 	}
-	updateTable, _ := upd.Table.Expr.(sqlparser.TableName)
+	if len(upd.TableExprs) > 1 {
+		return nil, errors.New("unsupported: multi-table update")
+	}
+	aliased, ok := upd.TableExprs[0].(*sqlparser.AliasedTableExpr)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
+	updateTable, ok := aliased.Expr.(sqlparser.TableName)
+	if !ok {
+		return nil, errors.New("unsupported: complex table reference in update")
+	}
```

Comment:
Does this work on joins? We are type asserting to type `*route` but I believe a query like `update foo from foo join bar on foo.id = bar.id` would be of type `join`.

Also what should I do with the `bldr` after I assert this. I believe the next step would be to Push things onto it, but maybe we don't need that for update/delete?